From: Yehuda Sadeh <yehuda@redhat.com>
Date: Thu, 29 Sep 2016 18:35:59 -0700
Subject: rgw: set correct instance on the object

Fixes: http://tracker.ceph.com/issues/17443

This was broken by commit bc840afafdfe5e528e5c1b711e71420ac3cb5a67

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>
(cherry picked from commit d2ada35c8c0d938a704ecf9974988ea30a9c0105)

Resolves (with prior): rhbz#1372346
---
 src/rgw/rgw_op.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rgw/rgw_op.cc b/src/rgw/rgw_op.cc
index 01adad8..4b3b928 100644
--- a/src/rgw/rgw_op.cc
+++ b/src/rgw/rgw_op.cc
@@ -264,7 +264,7 @@ static int read_policy(RGWRados *store, struct req_state *s,
     obj.set_in_extra_data(true);
   } else {
     obj = rgw_obj(bucket, object.name);
-    obj.set_instance(s->object.instance);
+    obj.set_instance(object.instance);
   }
   int ret = get_policy_from_attr(s->cct, store, s->obj_ctx, bucket_info, bucket_attrs, policy, obj);
   if (ret == -ENOENT && !object.empty()) {
